src/static/containers/AdminCardAdd/index.js

```diff
@@ -57,59 +56,133 @@ class AdminCardAddView extends Component {
     render = () => (
         <div className="protected">
             <AdminMenu />
-            <div className="col-lg-9">
-                <form onSubmit={this.createCard}>
-                    <Form ref={(ref) => { this.addCardForm = ref; }}
-                        type={Card}
-                        options={CardAddFormOptions}
-                        value={this.state.formValues}
-                        onChange={this.onFormChange}
-                    />
-                    <button type="submit" className="btn btn-success col-lg-4 col-lg-offset-4 col-xs-12">
-                        Create Card!
-                    </button>
-                </form>
-            </div>
+            {(this.props.isFetchingCard === true) ?
+                <p className="text-center">Loading card to edit...</p>
+                :
+                <div className="col-lg-9">
+                    <form onSubmit={this.createCard}>
+                        <Form ref={(ref) => { this.addCardForm = ref; }}
+                            type={Card}
+                            options={CardAddFormOptions}
+                            value={this.state.formValues}
+                            onChange={this.onFormChange}
+                        />
+                        <button type="submit" className="btn btn-success col-lg-4 col-lg-offset-4 col-xs-12">
+                            { this.state.editing ? 'Edit card!' : 'Create Card!'}
+                        </button>
+                    </form>
+                </div>
+            }
         </div>
     );
 }
 
+AdminCardAddView.defaultProps = {
+    match: null,
+    card_data: null
+};
+
 AdminCardAddView.propTypes = {
     token: PropTypes.string.isRequired,
+    isFetchingCard: PropTypes.bool.isRequired,
+    match: PropTypes.shape({
+        params: PropTypes.shape({
+            slug: PropTypes.string
+        })
+    }),
     actions: PropTypes.shape({
-        createCard: PropTypes.func.isRequired
+        createCard: PropTypes.func.isRequired,
+        editCard: PropTypes.func.isRequired,
+        cardFetch: PropTypes.func.isRequired
     }).isRequired,
+    card_data: PropTypes.shape({
+        title: PropTypes.string,
+        description: PropTypes.string,
+        image: PropTypes.string,
+        waste_reduction_score: PropTypes.number,
+        difficulty_score: PropTypes.number,
+        cost_score: PropTypes.number,
+        help_links: PropTypes.arrayOf(PropTypes.string),
+        card_stats: PropTypes.shape({
+            waste_reduction: PropTypes.number,
+            co2_reduction: PropTypes.number,
+            water_use_reduction: PropTypes.number,
+            status: PropTypes.string,
+            year: PropTypes.number,
+            data_sources: PropTypes.arrayOf(
+                PropTypes.shape({
+                    name: PropTypes.string,
+                    link: PropTypes.string,
+                    status: PropTypes.string
+                })
+            )
+        }),
```

Comment:
For PropTypes, you normally don't have to describe the whole object (think about a complex object coming from another lib).
Just the data / keys you use in your component